test/incidence.jl

```diff
@@ -0,0 +1,240 @@
+
+    incidence = Incidence(1)
+    @test incidence.typ === Const(0.0)
+    @test dependencies(incidence.row) == [2 => 1]
+    @test repr(incidence) == "Incidence(u₁)"
+
+    incidence = Incidence(3)
+    @test dependencies(incidence.row) == [4 => 1]
+    @test repr(incidence) == "Incidence(u₃)"
+
+    incidence = Incidence(Const(3.0), IncidenceValue[0.0, 0.0, 2.0, 1.0])
+    @test repr(incidence) == "Incidence(3.0 + 2.0u₂ + u₃)"
+
+    incidence = Incidence(Const(0.0), IncidenceValue[4.0, 0.0, 2.0])
+    @test repr(incidence) == "Incidence(4.0t + 2.0u₂)"
+
+    incidence = Incidence(Const(0.0), IncidenceValue[nonlinear])
+    @test repr(incidence) == "Incidence(f(t))"
+
+    incidence = Incidence(Const(0.0), IncidenceValue[linear])
+    @test repr(incidence) == "Incidence(cₜ * t)"
+
+    incidence = Incidence(Const(0.0), IncidenceValue[1.0, nonlinear])
+    @test repr(incidence) == "Incidence(t + f(u₁))"
+
+    incidence = Incidence(Const(0.0), IncidenceValue[1.0, linear])
+    @test repr(incidence) == "Incidence(t + c₁ * u₁)"
+
+    incidence = Incidence(Const(0.0), IncidenceValue[linear, linear, linear])
+    @test repr(incidence) == "Incidence(cₜ * t + c₁ * u₁ + c₂ * u₂)"
+
+    incidence = Incidence(Const(0.0), IncidenceValue[linear_state_dependent, linear_time_dependent, linear])
+    @test repr(incidence) == "Incidence(u₂ + f(∝t, ∝u₁))"
+
+    incidence = Incidence(Const(0.0), IncidenceValue[linear_state_dependent, linear_time_dependent, nonlinear])
+    @test repr(incidence) == "Incidence(f(∝t, ∝u₁, u₂))"
+
+    incidence = Incidence(Const(0.0), IncidenceValue[nonlinear, linear_time_dependent, nonlinear])
+    @test repr(incidence) == "Incidence(f(t, ∝u₁, u₂))"
+
+    @test_throws "inconsistent with an absence of time incidence" Incidence(Const(0.0), IncidenceValue[0.0, linear_time_dependent])
+    @test_throws "inconsistent with an absence of state incidence" Incidence(Const(0.0), IncidenceValue[linear_state_dependent])
+    @test_throws "absence of state dependence for time" Incidence(Const(0.0), IncidenceValue[1.0, linear_time_dependent])
+    @test_throws "absence of state dependence for time" Incidence(Const(0.0), IncidenceValue[linear, linear_time_dependent])
+  end
+
+  incidence = @incidence t
+  @test incidence.typ === Const(0.0)
+  @test dependencies(incidence.row) == [1 => 1]
+  @test repr(incidence) == "Incidence(t)"
+
+  incidence = @incidence 5. +ᵢ u₁
+  @test incidence.typ === Const(5.0)
+  @test dependencies(incidence.row) == [2 => 1]
+  @test repr(incidence) == "Incidence(5.0 + u₁)"
+
+  incidence = @incidence 5.0 +ᵢ t +ᵢ u₁
+  @test incidence.typ === Const(5.0)
+  @test dependencies(incidence.row) == [1 => 1, 2 => 1]
+  @test repr(incidence) == "Incidence(5.0 + t + u₁)"
+
+  incidence = @incidence t *ᵢ u₁
+  @test incidence.typ === Const(0.0)
+  @test dependencies(incidence.row) == [1 => linear_state_dependent, 2 => linear_time_dependent]
+  @test repr(incidence) == "Incidence(f(∝t, ∝u₁))"
```

Comment:
This is a tricky one, I think this one should probably just be `f₁(t) * u₁`

Reply:
As suggested in that issue, we could uppercase this macro and use it for printing the lattice element.

Reply:
Seems like it should be `f₁(t) * u₁ + f₂(u₁) * t` to cover, e.g., `@incidence t *ᵢ u₁ + t + u₁` which has the same incidence

Reply:
We don't have enough precision to resolve `f₂(u₁)`.

Reply:
For this special case we could resolve it (there is only one state, so state dependence means we depend on that single state), but this would break down as soon as we introduce more states. We'd need to track the full matrix to do that reliably.

Reply:
Yeah sorry the shorthand was a bit over-precise - the main suggestion is not to break symmetry here in the `repr`

Reply:
I that the intent would be to print time as a state coefficient, rather than state as a time coefficient. The symmetry would be broken by the fact that state is a somewhat more central as incidence information than time.

That does seem reasonable to me, but I'm also fine with the current printing.